### PR TITLE
Add working vSphere template

### LINKED ISSUE #328: Check that vSphere examples are up to date
Check that vSphere examples we provide are up to date

### PR DESCRIPTION
**Pull request by alexander-demicev**
**What this PR does / why we need it**:

This PR adds some fixes to make vSphere template work and merges all manifests in one template. Adding ClusterClass will be addressed later once we verify ClusterClasses work well in CAPRKE2.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #328

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: samples/vmware/README.md [hunk 1]
```diff
@@ -1,39 +1,39 @@
...
+VSPHERE_RESOURCE_POOL: "*/Resources"                          # The vSphere resource pool for your VMs
+VSPHERE_FOLDER: "vm"                                          # The VM folder for your VMs. Set to "" to use the root vSphere folder
+VSPHERE_TEMPLATE: "ubuntu-1804-kube-v1.17.3"                  # The VM template to use for your management cluster.
+CONTROL_PLANE_ENDPOINT_IP: "192.168.9.230"                    # the IP that kube-vip is going to use as a control plane endpoint
+VSPHERE_TLS_THUMBPRINT: "..."                                 # sha256 thumbprint of the vcenter certificate: openssl x509 -sha256 -fingerprint -in ca.crt -noout
+EXP_CLUSTER_RESOURCE_SET: "true"                              # This enables the ClusterResourceSet feature that we are using to deploy CSI
+VSPHERE_SSH_AUTHORIZED_KEY: "ssh-rsa AAAAB3N..."              # The public ssh authorized key on all machines in this cluster.
+                                                              #  Set to "" if you don't want to enable SSH, or are using another solution.
+"CPI_IMAGE_K8S_VERSION": "v1.25.0"                            # The version of the vSphere CPI image to be used by the CPI workloads
```
Comment (Danil-Grigorev) [inline samples/vmware/README.md:31]: Maybe it should use latest kubernetes version - 1.30?

File: samples/vmware/README.md [hunk 2]
```diff
@@ -1,39 +1,39 @@
...
-export CABPR_VCENTER_RESOURCEPOOL="*/Resources/resoucrepool"
-export CABPR_VCENTER_VM_VPCU=2
-export CABPR_VCENTER_VM_MEMORY=4096
-export CABPR_VCENTER_VM_TEMPLATE=template
+To initialize Cluster API Provider vSphere, clusterctl requires the following variables, which should be set in ~/.cluster-api/clusterctl.yaml as the following:
+
+```bash
+## -- Controller settings -- ##
+VSPHERE_USERNAME: "vi-admin@vsphere.local"                    # The username used to access the remote vSphere endpoint
+VSPHERE_PASSWORD: "admin!23"                                  # The password used to access the remote vSphere endpoint
```
Comment (Danil-Grigorev) [inline samples/vmware/README.md:16]:
```suggestion
VSPHERE_PASSWORD: "<password>"                                  # The password used to access the remote vSphere endpoint
```

**Review by Danil-Grigorev — DISMISSED** [review]
Only a couple of nits

File: samples/vmware/README.md [hunk 3]
```diff
@@ -1,39 +1,39 @@
...
 
-Create the namespace first.
+Then run the following command to generate the RKE2 cluster manifests:
 
-run:
-```shell
-envsubt < namespace.yaml | kubectl apply -f -
-envsubt < *.yaml | kubectl apply -f -
+```bash
+clusterctl generate cluster --from https://github.com/rancher/cluster-api-provider-rke2/blob/main/samples/vmware/cluster-template.yaml -n example-vsphere rke2-vsphere > vsphere-rke2-clusterctl.yaml
```
Comment (furkatgofurov7) [inline samples/vmware/README.md:38]: is it missing:

`kubectl apply -f vsphere-rke2-clusterctl.yaml`

after this?

File: samples/vmware/README.md [hunk 4]
```diff
@@ -1,39 +1,39 @@
...
-export CABPR_VCENTER_FOLDER=vm-folder
-export CABPR_VCENTER_RESOURCEPOOL="*/Resources/resoucrepool"
-export CABPR_VCENTER_VM_VPCU=2
-export CABPR_VCENTER_VM_MEMORY=4096
-export CABPR_VCENTER_VM_TEMPLATE=template
+To initialize Cluster API Provider vSphere, clusterctl requires the following variables, which should be set in ~/.cluster-api/clusterctl.yaml as the following:
+
+```bash
+## -- Controller settings -- ##
+VSPHERE_USERNAME: "vi-admin@vsphere.local"                    # The username used to access the remote vSphere endpoint
```
Comment (furkatgofurov7) [inline samples/vmware/README.md:15]:
```suggestion
VSPHERE_USERNAME: "<username>"                    # The username used to access the remote vSphere endpoint
```

File: samples/vmware/README.md (hunk 2 above)
Comment (alexander-demicev) [reply samples/vmware/README.md:16]: this is copy-paste from CAPV docs :D

**Review by furkatgofurov7 — DISMISSED** [review]

**Review by furkatgofurov7 — APPROVED** [review]

**Comment by furkatgofurov7** [comment]
e2e tests are failing, but that is unrelated (e2e failures on main as well) and changes of this PR can't be tested even they were green. This could be merged as is IMO.

**Review by Danil-Grigorev — APPROVED** [review]

**Pull request merged**